code/src/cljs/sixsq/nuvla/ui/credentials/utils.cljs

```diff
@@ -93,13 +93,93 @@
+
+
+(defn db->new-amazonec2-credential
+  [db]
+  (let [name        (get-in db [::spec/credential :name])
+        description (get-in db [::spec/credential :description])
+        subtype     (get-in db [::spec/credential :subtype])
+        key         (get-in db [::spec/credential :amazonec2-access-key])
+        secret      (get-in db [::spec/credential :amazonec2-secret-key])
+        acl         (get-in db [::spec/credential :acl])]
+    (-> {}
+        (assoc :name name)
+        (assoc :description description)
+        (assoc-in [:template :href] (str "credential-template/store-" subtype))
+        (assoc-in [:template :amazonec2-access-key] key)
+        (assoc-in [:template :amazonec2-secret-key] secret)
+        (cond-> acl (assoc-in [:template :acl] acl)))))
+
+
+(defn db->new-azure-credential
+  [db]
+  (let [name        (get-in db [::spec/credential :name])
+        description (get-in db [::spec/credential :description])
+        subtype     (get-in db [::spec/credential :subtype])
+        subsciption (get-in db [::spec/credential :azure-subscription-id])
+        id          (get-in db [::spec/credential :azure-client-id])
+        secret      (get-in db [::spec/credential :azure-client-secret])
+        acl         (get-in db [::spec/credential :acl])]
+    (-> {}
+        (assoc :name name)
+        (assoc :description description)
+        (assoc-in [:template :href] (str "credential-template/store-" subtype))
+        (assoc-in [:template :azure-subscription-id] subsciption)
+        (assoc-in [:template :azure-client-id] id)
+        (assoc-in [:template :azure-client-secret] secret)
+        (cond-> acl (assoc-in [:template :acl] acl)))))
+
+
+(defn db->new-google-credential
+  [db]
+  (let [name        (get-in db [::spec/credential :name])
+        description (get-in db [::spec/credential :description])
+        subtype     (get-in db [::spec/credential :subtype])
+        username    (get-in db [::spec/credential :google-username])
+        project     (get-in db [::spec/credential :google-project])
+        id          (get-in db [::spec/credential :client-id])
+        secret      (get-in db [::spec/credential :client-secret])
+        token       (get-in db [::spec/credential :refresh-token])
+        acl         (get-in db [::spec/credential :acl])]
+    (-> {}
+        (assoc :name name)
+        (assoc :description description)
+        (assoc-in [:template :href] (str "credential-template/store-" subtype))
+        (assoc-in [:template :google-username] username)
+        (assoc-in [:template :google-project] project)
+        (assoc-in [:template :client-id] id)
+        (assoc-in [:template :client-secret] secret)
+        (assoc-in [:template :refresh-token] token)
+        (cond-> acl (assoc-in [:template :acl] acl)))))
+
+
 (defn db->new-credential
   [db]
   (let [subtype (get-in db [::spec/credential :subtype])]
     (case subtype
       "infrastructure-service-swarm" (db->new-swarm-credential db)
```

Comment:
It miss support for infrastructure-service-kubernetes

Reply:
This is addressed with this commit 9d7764d